core/deployment/src/main/java/io/quarkus/runner/RuntimeRunner.java

```diff
@@ -115,6 +116,12 @@ public void run() {
 
                 transformerTarget.setTransformers(functions);
             }
+            for (GeneratedClassBuildItem i : result.consumeMulti(GeneratedClassBuildItem.class)) {
```

Comment:
Maybe these should be in a build step too?

Reply:
What do you mean? I think generated classes are a reasonable output from the runtime runners point of view, I don't think trying to stuff them into a class loader as part of a build step feels right.

Reply:
Previously they just wrote to the class output, which was a build item.  If class output were still a build item, these could be in steps and duplication removed.

Reply:
ClassOutputBuildItem violated the BuildItem contract though. It was a build item that was consumed to produce output. After the initial set of changes the build step that produced the main class ended up being ignored, as it was only consuming ClassOutputBuildItem and not producing something the jar build steps expected.

If something is used to produce output it needs to be a BuildProducer

Reply:
But it already existed, which is a big plus. :)

Really having something that is a build item that "catches" output isn't all that bad - it's essentially similar to a build item that specifies where to put something; it's just a bit more active about it.  So in this case the "output" is the fact that the classes were produced, which can be represented by a virtual build item.

Reply:
But it's broken, it only worked by accident because it was consumed by things that were also produced things necessary for the build.

You can make it work by always consuming it in a build step that also produces something useful, but that is pretty hacky.

Reply:
With no outputs it may have worked on accident in the past, but by wiring it in to a virtual build output it stops being hacky IMO.  But it's not super important to resolve now, I was just thinking of a way to encapsulate things a little better.